pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -86,7 +86,7 @@ func (u UsernamePassword) Setup(baseDir string) error {
 	fmt.Fprintf(gitconfig, UserPassGitConfig, gitcredentials.Name())
 	fmt.Fprintf(gitcredentials, "%s", usernamePasswordURL.String())
 
-	return ensureGitConfigIncludes(gitconfig.Name())
+	return &usernamePasswordURL, ensureGitConfigIncludes(gitconfig.Name())
```

Comment:
Actually the URL returned should be the URL without Username/Pass. That way the credentials are not output in the logs.
git will then use the credential store and the gitcredentials file to figure out the credentials to use.